### DISCUSSION TITLE
[Chapter 12] 순서가 필요한 구독 메서드 예시는 어떤 것이 있을까?

### DISCUSSION THREAD

**Discussion by corock** [cowaine/spring-boot-for-msa#195] [Chapter 12. Spring Events] [answered]
12.4 `@Async` 애너테이션을 사용한 비동기 이벤트 처리 장에서 `HotelEventListener` 코드를 설명하기 전에 다음과 같은 구절이 나옵니다.

> 또한 두 개 이상의 구독 메서드가 동작할 때 순서가 필요하다면 `@Order` 애너테이션을 사용한다.

순서가 필요한 구독 메서드 예시가 어떤 것이 있을까요?

**Comment by ahngilwoong** [comment]
하나의 예시로는 db에 저장되는 이벤트와 그 저장된 디비 정보를 사용해야하는 이벤트가 있으면 순서가 필요할거라고 생각이 듭니다.
즉 의존적인 이벤트들끼리는 순서가 필요한 것 같습니다.

**Comment by Irisation23** [comment, marked as answer]
### 회원가입

1. Order(1) 이메일 인증
2. 회원가입 버튼 -> Submit
3. 해당 아이디에 포인트 지급 Order(2)
